### DISCUSSION TITLE
permission denied on Werf container image

### DISCUSSION THREAD

**Discussion by ahmed-dabouni** [werf/werf#6199] [General]
Hello everyone,

I'm working on a project using a Docker Executor for GitLab Runner and Werf Container Image to build and deploy a Spring Boot application.

The script runs as the "**build**" user and cannot write to the "**/home/build/.werf**" directory because its permissions are set to "root:root". 

![2024-06-10 02_49_54-Build and Publish (#8280857) · Jobs · swe2_if_2024 _ projekt_01 · GitLab — Mozil](https://github.com/werf/werf/assets/92936391/4e914aad-8b13-41fa-99bd-d75980d595be)

The commented line in the pipeline resolved this issue temporarily. Could you please adjust the permissions to "build:build" in the upcoming container image release? Or am I perhaps missing something that's causing this error? Any guidance would be greatly appreciated.

![2024-06-10 02_49_28-projekt_01 –  gitlab-ci yml  projekt_01](https://github.com/werf/werf/assets/92936391/a26b6dd1-4119-449d-a5a7-d408cc6a683e)

Thanks, Ahmed

**Comment by ilya-lesikov** [comment]
You probably have this in your runner config:
```toml
    volumes = ["/home/build/.werf"]
```

And it looks like if this directory does not exist in the image yet, then it will be created with uid/gid 0. But ownership of the directory will be preserved if it was already present in the image.

So we need to create this directory during building of our images, possibly should be done for `/builds` also.

**Reply by ahmed-dabouni** [reply]
Yes, exactly. I have this line in my runner config:

`volumes = ["/home/build/.werf"]`

I have followed this official guide from the Werf website under the "[Configuring GitLab Runner](https://werf.io/getting_started/cicd/gitlabcicd-dockerrunner-linux-buildah-bestpractice-no-application.html#configuring-gitlab-runner)" section, and I have done exactly as described.

**Comment by ilya-lesikov** [comment]
Will be tracked in #6200

**Comment by alexey-igrychev** [comment]
fixed https://github.com/werf/werf/pull/6382
